client/src/components/Chat.jsx

```diff
@@ -323,89 +320,145 @@ const Chat = () => {
 					className="h-[100%] md:max-h-full overflow-y-auto w-full scroll-smooth"
 				>
 					{sortedMessages.map(
-						({ senderId: sender, id, message, time, status, isEdited, oldMessages, containsBadword, isRead }) => {
+						({ senderId: sender, id, message, time, status, isEdited, oldMessages, containsBadword, isRead, replyTo }) => {
 							const isSender = sender.toString() === senderId.toString();
 							message = decryptMessage(message)
+							// original message this message is a reply to
+							const repliedMessage = replyTo ? (() => {
+								const messageObj = getMessage(replyTo)
+								if (!messageObj) {
+									return null
+								}
+
+								return {
+									...messageObj,
+									message: decryptMessage(messageObj.message)
+								}
+							})() : null
+
+							// is this message currently being replied? 
+							const hasActiveReply = currentReplyMessageId === id
 
 							return (
-								<div
-									key={id}
-									id={!isSender && `message-${id}`}
-									className={`w-full flex text-white relative mb-2 ${
-										isSender ? 'justify-end' : 'justify-start'
-									}`}
-								>
+								<div key={id} id={`message-${id}`} className={createClassesFromArray(
+									'flex flex-col gap-2 py-2 duration-500 transition-all',
+									hasActiveReply && 'bg-[#FF9F1C]/25 border-[#FF9F1C]',
+									hasActiveReply && (isSender ? 'border-r-[3.5px]' : 'border-l-[3.5px]')
+								)}>
```

Comment:
please use the regular way of tailwind here

Reply:
Can you elaborate? I'm obviously using tailwind classes

Reply:
CreateClassesArray is what I mean, don't use that

Reply:
Sorry, but how does that affect anything?
I used it because there were multiple conditional classes, and putting them in a template string makes the code unreadable. Using the util makes prettier format the classes used. Plus the util is part of the codebase.